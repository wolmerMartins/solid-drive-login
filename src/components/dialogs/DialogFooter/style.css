.DialogFooter {
  margin-top: 1rem;
  text-align: center;
  font-size: var(--xxs-font);
}
