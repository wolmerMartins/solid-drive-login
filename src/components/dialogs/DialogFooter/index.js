import React, { forwardRef } from 'react'

import './style.css'

const DialogFooter = forwardRef(({ children = null }, ref) => {
  const setRef = el => {
    if (!ref) return

    ref.current.footer = el
  }

  return (
    <footer
      ref={setRef}
      className="DialogFooter"
    >
      {children}
    </footer>
  )
})

export default DialogFooter
