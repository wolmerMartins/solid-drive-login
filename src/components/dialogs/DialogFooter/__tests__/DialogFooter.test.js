import React, { useRef } from 'react'
import { render } from '@testing-library/react'

import DialogFooter from '../index'

const TestComponent = ({ withRef, children }) => {
  const elRef = useRef({})

  return (
    <DialogFooter ref={withRef ? elRef : null}>
      {children}
    </DialogFooter>
  )
}

const buildComponent = (props, withRef) => {
  const utils = render(<TestComponent withRef={withRef} {...props} />)
  const footer = utils.queryByRole('contentinfo')

  return {
    ...utils,
    footer
  }
}

describe('<DialogFooter />', () => {
  it('Render the component empty', () => {
    const { footer } = buildComponent()

    expect(footer.firstElementChild)
      .not.toBeInTheDocument()
  })

  it('Render the component with a children', () => {
    const { footer } = buildComponent({ children: <span>Dialog Footer</span> })

    expect(footer)
      .toBeInTheDocument()

    expect(footer.textContent)
      .toContain('Dialog Footer')
  })

  it('Render the component with ref', () => {
    const { asFragment } = buildComponent({ children: <span>Dialog Footer</span> }, true)

    expect(asFragment())
      .toMatchSnapshot()
  })
})
