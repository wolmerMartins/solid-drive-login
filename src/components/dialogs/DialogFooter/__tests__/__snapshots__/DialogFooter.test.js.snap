// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogFooter /> Render the component with ref 1`] = `
<DocumentFragment>
  <footer
    class="DialogFooter"
  >
    <span>
      Dialog Footer
    </span>
  </footer>
</DocumentFragment>
`;
